src/pools/cluster/dynamic.ts

```diff
@@ -61,7 +61,8 @@ export class DynamicClusterPool<
     // All workers are busy, create a new worker
     const worker = this.createAndSetupWorker()
     this.registerWorkerMessageListener<Data>(worker, message => {
-      if (message.kill) {
+      const tasksInProgress = this.tasks.get(worker)
+      if (message.kill && !tasksInProgress) {
```

Comment:
We shouldn't make such a check, lets better use 
```suggestion
      if (message.kill && tasksInProgress === 0) {
```
to make it more readable

Reply:
I was thinking the same I  will do